lib/matplotlib/ticker.py

```diff
@@ -1550,11 +1550,9 @@ def view_limits(self, vmin, vmax):
             vmax += 1
 
         if rcParams['axes.autolimit_mode'] == 'round_numbers':
-            exponent, remainder = _divmod(math.log10(vmax - vmin),
-                                         math.log10(max([self.numticks-1, 1])))
-            if remainder < 0.5:
-                exponent -= 1
-            scale = max([self.numticks-1, 1]) ** (-exponent)
+            exponent = round(math.log10(vmax - vmin)
+                             / math.log10(max(self.numticks - 1, 1))) - 1
```

Comment:
You are correct that this is different: the previous implementation compares the remainder with 0.5 whereas I thought it was comparing it with half the divisor.  My guess is that the previous version was incorrect.

However I can also just revert to the previous version...